### PR TITLE
[inference Zero-Dim] trt support 0 dims

### PR DESCRIPTION
**Pull request by zhangjun**
### PR types

Others
### PR changes

Others
### Description

- tensorrt support 0 dims
- TODO：OP support one by one

### CODE REVIEW

File: paddle/fluid/operators/tensorrt/tensorrt_engine_op.h [hunk 1]
```diff
@@ -64,19 +65,10 @@ using inference::tensorrt::TRTInt8Calibrator;
 
 static void RuntimeStaticShapeCheck(std::vector<int64_t> runtime_input_shape,
                                     std::vector<int64_t> model_input_shape) {
-  auto comma_fold = [](std::string a, int b) {
-    return std::move(a) + ", " + std::to_string(b);
-  };
   std::string model_input_shape_str =
-      std::accumulate(std::next(model_input_shape.begin()),
-                      model_input_shape.end(),
-                      std::to_string(model_input_shape[0]),
```
Comment (zhangjun) [inline paddle/fluid/operators/tensorrt/tensorrt_engine_op.h:73]: 0 dim，这里会报错

File: test/ir/inference/test_trt_convert_activation.py [hunk 2]
```diff
@@ -33,7 +33,9 @@ def is_program_valid(self, program_config: ProgramConfig) -> bool:
 
     def sample_program_configs(self):
         def generate_input1(dims, batch, attrs: List[Dict[str, Any]]):
-            if dims == 1:
+            if dims == 0:
+                return np.random.random([]).astype(np.float32)
```
Comment (zhangjun) [inline test/ir/inference/test_trt_convert_activation.py:37]: 增加0维单测

File: test/ir/inference/test_trt_convert_activation.py [hunk 3]
```diff
@@ -132,7 +144,7 @@ def clear_dynamic_shape():
             self.dynamic_shape.opt_input_shape = {}
 
         def generate_trt_nodes_num(attrs, dynamic_shape):
-            if self.dims == 1:
+            if not dynamic_shape and (self.dims == 1 or self.dims == 0):
```
Comment (zhangjun) [inline test/ir/inference/test_trt_convert_activation.py:147]: Static shape does not support 0 dim Tensor.

File: paddle/fluid/inference/tensorrt/op_teller.cc [hunk 4]
```diff
@@ -116,9 +116,10 @@ struct SimpleOpTypeSetTeller : public Teller {
       auto x_var_name = desc.Input("X")[0];
       auto* x_var_desc = block->FindVar(x_var_name);
       const auto x_shape = x_var_desc->GetShape();
-      if (x_shape.size() == 1) {
+      if (!with_dynamic_shape && (x_shape.size() == 1 || x_shape.size() == 0)) {
```
Comment (zhangjun) [inline paddle/fluid/inference/tensorrt/op_teller.cc:119]: Static shape does not support 0 dim Tensor.

**Review by qingqing01 — APPROVED** [review]

**Review by zhwesky2010 — APPROVED** [review]

**Pull request merged**